satellite/metainfo/metainfo.go

```diff
@@ -511,20 +514,37 @@ func CreatePath(ctx context.Context, projectID uuid.UUID, segmentIndex int64, bu
 func (endpoint *Endpoint) SetAttribution(ctx context.Context, req *pb.SetAttributionRequest) (_ *pb.SetAttributionResponse, err error) {
 	defer mon.Task()(&ctx)(&err)
 
-	_, err = endpoint.checkBucketPointers(ctx, req)
+	projectID, err := endpoint.checkBucketPointers(ctx, req)
 	if err != nil {
-		// TODO: return correct status code for GRPC
-		endpoint.log.Sugar().Debug("related bucket id already attributed \n")
-		return &pb.SetAttributionResponse{}, err
+		return nil, err
 	}
 
-	// TODO: add valueattribution DB access functions added in new PR.
-
+	// check if already bucket exists
+	_, err = endpoint.partnerinfo.Get(ctx, *projectID, req.GetBucketName())
```

Comment:
I feel that logic in SetAttribution is a little bit awkward. I believe it would be more optimal to have such steps:
* check if attribution is set for given bucket
* return error if yes
* check if bucket exists 
* return error if no
* check if bucket is empty
* return error if no
* set attribution

Reply:
refactored the handling of bucket attribution as suggested